src/main/java/com/whichclasses/statistics/ConcreteAggregator.java

```diff
@@ -0,0 +1,35 @@
+package com.whichclasses.statistics;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.Maps;
+import com.google.inject.Inject;
+import com.whichclasses.model.Course;
+import com.whichclasses.model.DataSource;
+import com.whichclasses.model.Department;
+import com.whichclasses.model.proto.TceRating.Question;
+
+public class ConcreteAggregator implements DataAggregator {
```

Comment:
That seems reasonable, what would the return type be though? Why not always
a Rateable?

On Tue, May 5, 2015 at 12:09 PM, Sean Topping <notifications@github.com>
wrote:

> In src/main/java/com/whichclasses/statistics/ConcreteAggregator.java
> <https://github.com/cleichner/whichclasses/pull/3#discussion_r29702022>:
>
> > +package com.whichclasses.statistics;
> > +
> > +import java.util.ArrayList;
> > +import java.util.Collection;
> > +import java.util.Collections;
> > +import java.util.List;
> > +import java.util.Map;
> > +
> > +import com.google.common.collect.Maps;
> > +import com.google.inject.Inject;
> > +import com.whichclasses.model.Course;
> > +import com.whichclasses.model.DataSource;
> > +import com.whichclasses.model.Department;
> > +import com.whichclasses.model.proto.TceRating.Question;
> > +
> > +public class ConcreteAggregator implements DataAggregator {
>
> Maybe we can change DataAggregator so that it just implements a single
> "aggregate" method. This way we can have sorting aggregators, grouping
> aggregators, etc. Maybe genericize DataAggregator with the return type of
> the aggregate method?
>
> public class RankingDataAggregator implements DataAggregator>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cleichner/whichclasses/pull/3/files#r29702022>.
>